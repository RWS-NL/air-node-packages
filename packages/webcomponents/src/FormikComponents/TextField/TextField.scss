@import '../../styles/vars';

.outline {
  border-radius: 0 !important;
  border-width: 1px;
}

.input {
  color: $textBlack;
  font-size: 0.9rem;
  height: 2.5rem;
  min-height: 2.5rem;
}

.inputLabel {
  font-size: 1rem;
  font-style: italic;
  transform: translate(14px, -6px) scale(0.75);
}

.error {
  &Label {
    color: $red100;
    position: absolute;
    top: 2.5rem;
  }

  &Input {
    :hover > fieldset {
      border-width: 2px;
    }
  }
}
