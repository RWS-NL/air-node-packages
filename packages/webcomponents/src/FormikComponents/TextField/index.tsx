import { TextField as MUITextField } from '@material-ui/core';
import classnames from 'classnames';
import { FieldHookConfig, useField } from 'formik';
import { TextFieldProps, useFieldToTextField } from 'formik-material-ui';
import React, { FC } from 'react';
import css from './TextField.scss';

/**
 * Constructs a TextField with the formik validation
 * @param props Props to pass to the Textield component
 * @example
 * ```jsx
 * <TextField
 *  name='name' type='text' label='Example'
 *  placeholder='Example placeholder' variant='outlined'
 *  data-qa='exampleDataQA' required
 * />
 * ```
 */
export const TextField: FC<TextFieldProps> = props => {
  const [{ value, onChange, onBlur }, { error, touched }] = useField(props as FieldHookConfig<any>);
  const textFieldProps = useFieldToTextField(props);

  const textFieldHasErrors = Boolean(error) && touched;

  return (
    <MUITextField
      {...props}
      {...textFieldProps}
      value={value}
      label={props.label}
      placeholder={props.placeholder}
      onChange={onChange}
      onBlur={onBlur}
      margin='dense'
      variant='outlined'
      className={props.className}
      fullWidth
      autoFocus={props.autoFocus}
      required={props.required}
      InputProps={{
        classes: {
          root: css.input,
          notchedOutline: css.outline
        }
      }}
      classes={{ root: classnames([{ [css.errorInput]: textFieldHasErrors }]) }}
      InputLabelProps={{
        shrink: true,
        classes: { root: css.inputLabel }
      }}
      FormHelperTextProps={{
        component: 'div',
        classes: { error: css.errorLabel }
      }}
    />
  );
};

export default TextField;
