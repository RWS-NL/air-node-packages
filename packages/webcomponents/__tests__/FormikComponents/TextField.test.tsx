import { shallow } from 'enzyme';
import { Formik } from 'formik';
import React from 'react';
import TextField from '../../src/FormikComponents/TextField';

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const textField = shallow(
      <Formik initialValues={{ name: '' }} onSubmit={() => undefined}>
        <TextField type='text' name='field' />
      </Formik>
    );
    expect(textField).toMatchSnapshot();
  });
});
