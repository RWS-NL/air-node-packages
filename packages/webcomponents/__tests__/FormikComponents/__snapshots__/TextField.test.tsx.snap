// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Required Props 1`] = `
<ContextProvider
  value={
    Object {
      "dirty": false,
      "errors": Object {},
      "getFieldHelpers": [Function],
      "getFieldMeta": [Function],
      "getFieldProps": [Function],
      "handleBlur": [Function],
      "handleChange": [Function],
      "handleReset": [Function],
      "handleSubmit": [Function],
      "initialErrors": Object {},
      "initialStatus": undefined,
      "initialTouched": Object {},
      "initialValues": Object {
        "name": "",
      },
      "isSubmitting": false,
      "isValid": true,
      "isValidating": false,
      "registerField": [Function],
      "resetForm": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFormikState": [Function],
      "setStatus": [Function],
      "setSubmitting": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "status": undefined,
      "submitCount": 0,
      "submitForm": [Function],
      "touched": Object {},
      "unregisterField": [Function],
      "validateField": [Function],
      "validateForm": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
      "validateOnMount": false,
      "values": Object {
        "name": "",
      },
    }
  }
>
  <TextField
    name="field"
    type="text"
  />
</ContextProvider>
`;
